hdr/sqlite_modern_cpp.h

```diff
@@ -399,41 +399,44 @@ namespace sqlite {
 		);
 	}
 	
+	enum class Encoding {
+	  AUTO = SQLITE_ANY,
+	  UTF8 = SQLITE_UTF8,
+	  UTF16 = SQLITE_UTF16
+	};
 	struct sqlite_config {
+    int flags = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
+    const char *gVfs = nullptr;
+    Encoding encoding = Encoding::AUTO;
 	};
 
 	class database {
 	protected:
 		std::shared_ptr<sqlite3> _db;
 
 	public:
-		database(std::u16string const & db_name): _db(nullptr) {
+		database(const std::string &db_name, const sqlite_config &config = {}): _db(nullptr) {
 			sqlite3* tmp = nullptr;
-			auto ret = sqlite3_open16(db_name.data(), &tmp);
+			auto ret = sqlite3_open_v2(db_name.data(), &tmp, config.flags, config.gVfs);
 			_db = std::shared_ptr<sqlite3>(tmp, [=](sqlite3* ptr) { sqlite3_close_v2(ptr); }); // this will close the connection eventually when no longer needed.
 			if(ret != SQLITE_OK) exceptions::throw_sqlite_error(ret);
-			//_db.reset(tmp, sqlite3_close); // alternative close. (faster?)
+			if(config.encoding == Encoding::UTF16)
+			  *this << R"(PRAGMA encoding = "UTF-16";)";
 		}
 
-		database(std::string const & db_name): _db(nullptr) {
+		database(const std::u16string &db_name, const sqlite_config &config = {}): _db(nullptr) {
```

Comment:
Make the default encoding for this constructor `UTF16`, this changes the behavior we had in previous versions.
@zauguin